portable/GCC/ARM_CM4F/port.c

```diff
@@ -543,23 +534,45 @@ void xPortSysTickHandler( void )
 		to be unsuspended then abandon the low power entry. */
 		if( eTaskConfirmSleepModeStatus() == eAbortSleep )

```

Comment:
I think this is a good optimization -- confirming sleep mode first, and only disable SysTick temporarily in the branch where tickless idle is needed. 👍

Reply:
Thank you.  It takes a little extra work to make it happen and some consideration of a race condition the code had previously avoided.  But I think it's worth doing especially when bug fixes are going into the same function.